pennylane/ops/op_math/sprod.py

```diff
@@ -76,13 +73,10 @@ def s_prod(scalar, operator, lazy=True, do_queue=None, id=None):
            [ 2., 0.]])
     """
     if lazy or not isinstance(operator, SProd):
-        return SProd(scalar, operator, do_queue=do_queue, id=id)
-
-    sprod_op = SProd(scalar=scalar * operator.scalar, base=operator.base, do_queue=do_queue, id=id)
-
-    if do_queue or do_queue is None:
-        QueuingManager.remove(operator)
+        return SProd(scalar, operator, id=id)
 
+    sprod_op = SProd(scalar=scalar * operator.scalar, base=operator.base, id=id)
+    QueuingManager.remove(operator)
```

Comment:
What happens here when `QueuingManager.recording() == False`? Not sure about this.
```suggestion
    if QueuingManager.recording():
        QueuingManager.remove(operator)
```

Reply:
it's handled [in QueuingManager.remove](https://github.com/PennyLaneAI/pennylane/blob/7bc6255ec1a8a43bed7688a772716486ade21087/pennylane/queuing.py#L324C10-L324C10) already 😄

Reply:
Oh, that's wonderful.